File: 2025-05-MAY-GFG-POTD/May 01 - Pascal Triangle/Solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> nthRowOfPascalTriangle(int n) {
        vector<int> row;
        int mod = 1000000007;

        long val = 1;
        row.push_back((int)val);

        for (int k = 1; k < n; k++) {
            val = val * (n - k);
            val = val / k;
            val %= mod;
            row.push_back((int)val);
        }

        return row;
    }
};

void printAns(vector<int>& ans) {
    for (int x : ans) {
        cout << x << " ";
    }
    cout << endl;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;

        Solution obj;
        vector<int> ans = obj.nthRowOfPascalTriangle(n);
        printAns(ans);
        cout << "~" << endl;
    }

    return 0;
}

File: 2025-05-MAY-GFG-POTD/May 01 - Pascal Triangle/Solution.py
MOD = 1000000007

class Solution:
    def nthRowOfPascalTriangle(self, n):
        row = []
        val = 1
        row.append(val)

        for k in range(1, n):
            val = val * (n - k)
            val = val // k
            val = val % MOD
            row.append(val)

        return row

# Driver code
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        n = int(input())
        sol = Solution()
        ans = sol.nthRowOfPascalTriangle(n)
        print(" ".join(map(str, ans)))
        print("~")
